Source/WebCore/html/HTMLElement.cpp

```diff
@@ -1399,14 +1405,18 @@ ExceptionOr<void> HTMLElement::showPopover()
         if (popoverState() != originalState)
             return Exception { InvalidStateError, "The value of the popover attribute was changed while hiding the popover."_s };
 
-        if (auto check = checkPopoverValidity(*this, PopoverVisibilityState::Hidden, document.ptr()); check.hasException())
+        check = checkPopoverValidity(*this, PopoverVisibilityState::Hidden, document.ptr());
+        if (check.hasException())
             return check.releaseException();
+        if (!check.returnValue())
+            return { };
     }
 
     bool shouldRestoreFocus = !document->topmostAutoPopover();
     RefPtr previouslyFocusedElement = document->focusedElement();
 
-    addToTopLayer();
+    if (!isInTopLayer())
+        addToTopLayer();
```

Comment:
Why is this change needed? I don't see it in the spec.

Reply:
Right, I was trying multiple things at this time, but for this patch this change is not needed, removed.